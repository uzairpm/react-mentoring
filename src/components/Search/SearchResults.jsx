import React, {Component} from 'react';

import NoResults from  './NoResults';
import MovieCards from './MovieCards';

export default class SearchResults extends Component {
    constructor(props) {
        super(props);
    }
    render() {
        return (
            <div className="container">
                { this.props.movies.length === 0 && <NoResults/> }
                { this.props.movies.length > 0 && <MovieCards
                    movies={this.props.movies} 
                    releaseDate={this.props.releaseDate}
                    releaseDateClick={this.props.releaseDateClick}
                    ratingClick={this.props.ratingClick} />}
            </div>
        );
    }
}
