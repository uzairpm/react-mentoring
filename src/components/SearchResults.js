import React, {Component} from 'react';

import Button from './Common/Button';
import MovieCard from './MovieCard';
import SearchFilter from './SearchFilter';
import Footer from './Footer/Footer';

import './searchresults.css';

export default class SearchResults extends Component {
    constructor(props) {
        super(props);
    }
    
    render() {
        const movieCards = this.props.movies.map(movie => {
            return <MovieCard key={movie.id} movie={movie} />
        });
        let view = '';
        if (this.props.movies.length === 0) {
            view = (
                <h1 className="display-2 text-center mt-5">No films found</h1>
            );
        } else {
            view = (
                <div>
                    <SearchFilter resultCount={this.props.movies.length}
                        releaseDate={this.props.releaseDate}
                        releaseDateClick={this.props.releaseDateClick}
                        ratingClick={this.props.ratingClick} />
                    <div className="row">
                        { movieCards }
                    </div>
                    <Footer />
                </div>
            );
        }
        return (
            <div className="container">
                {view}
            </div>
        );
    }
}