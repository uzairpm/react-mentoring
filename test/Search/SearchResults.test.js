import React from 'react';

import SearchResults from '../../src/components/Search/SearchResults';

describe('Check Component creation', () => {
    test('SearchResults component', () => {
        const moviesArray = [{},{}];
        const element = shallow(<SearchResults movies={moviesArray}/>);
        expect(element).toMatchSnapshot();
    });
    test('SearchResults component 2', () => {
        const moviesArray = [];
        const element2 = shallow(<SearchResults movies={moviesArray}/>);
        expect(element2).toMatchSnapshot();
    });
});
