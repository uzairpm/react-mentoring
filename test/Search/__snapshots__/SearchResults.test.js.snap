// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check Component creation SearchResults component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchResults
    movies={
      Array [
        Object {},
        Object {},
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        false,
        <MovieCards
          movies={
            Array [
              Object {},
              Object {},
            ]
          }
          ratingClick={undefined}
          releaseDate={undefined}
          releaseDateClick={undefined}
        />,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "movies": Array [
            Object {},
            Object {},
          ],
          "ratingClick": undefined,
          "releaseDate": undefined,
          "releaseDateClick": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          false,
          <MovieCards
            movies={
              Array [
                Object {},
                Object {},
              ]
            }
            ratingClick={undefined}
            releaseDate={undefined}
            releaseDateClick={undefined}
          />,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "movies": Array [
              Object {},
              Object {},
            ],
            "ratingClick": undefined,
            "releaseDate": undefined,
            "releaseDateClick": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Check Component creation SearchResults component 2 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchResults
    movies={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NoResults />,
        false,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NoResults />,
          false,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
